### PR TITLE
[chore][cmd/mdatagen] Properly test bytes attribute type

### LINKED ISSUE #29923: mdatagen: attribute with type `bytes` generates erroneous unit tests
### Component(s)

cmd/mdatagen

## Description
Defining an attribute in `metadata.yaml` with `bytes` type generates unit tests which do not compile. The [schema](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/9378ab5a15e4fe481fdc03627138e5def9638eb5/cmd/mdatagen/metadata-schema.yaml#L68) defines `bytes` as a valid type for an attribute.

However, the unit tests tries to create byte arrays like:
```go
mb.RecordMetric1(ts, bytes("attr1-val")
```

However, I believe it should be:
```go
mb.RecordMetric1(ts, []byte("attr1-val")
```

## Steps to Reproduce
Define an attribute in `metadata.yaml` as:
```yaml
attributes:
  attr1:
    description: An attribute with type bytes
    type: bytes

metrics:
  metric1:
  desciption: A metric
  enabled: true
  sum:
    value_type: int
    monotonic: true
  unit: kg
  attributes:
    - attr1
```

## Expected Result
A compiling unit test

## Actual Result
A not compiling unit test

### Collector version

v0.90.0

## Environment
OS: Ubuntu v22.04
Compiler(if manually compiled): go 1.21.3 linux/amd64

### OpenTelemetry Collector configuration

_No response_

### Log output

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by crobert-1**
**Description:** <Describe what has changed.>

Changes how attributes of type `bytes` are tested, as they were being initialized incorrectly before.

**Link to tracking Issue:** <Issue number if applicable>
Resolves #29923

**Testing:** <Describe what testing was performed and which tests were added.>
Fake attribute added for testing:
```
attributes:
  # Added a fake attribute in a metadata.yaml file
  fake:
    description: "Fake attribute for testing"
    type: bytes
```
Test case generated by `cmd/mdatagen`:
```
                                       attrVal, ok = dp.Attributes().Get("fake")
                                       assert.True(t, ok)
                                       assert.EqualValues(t, []byte("fake-val"), attrVal.Bytes())
```

### CODE REVIEW

File: cmd/mdatagen/loader.go
Comment (mmynk) [inline cmd/mdatagen/loader.go:197]: Is it possible to unit test these? Seems like a really basic thing which wasn't caught by the extensive CI.

Comment (crobert-1) [reply cmd/mdatagen/loader.go:197]: There aren't tests for this at this point. It's not a bad idea, but given the facts that this is only used for generating test code, it shouldn't change very frequently, and it can't build when something is wrong, I don't see as much value for this specific test case at this point. I'm open to feedback from others though.

**Comment by mmynk** [comment]
@crobert-1 possible to have it merged soon? seems like an important bug fix.

**Comment by crobert-1** [comment]
> @crobert-1 possible to have it merged soon? seems like an important bug fix.

We have to wait for proper approvals and then a maintainer will have to merge. Many people are still coming back from time off, so it may just take some time to get merged in. I'll ping people next week if there's no update 👍

**Comment by mmynk** [comment]
> > @crobert-1 possible to have it merged soon? seems like an important bug fix.
> 
> We have to wait for proper approvals and then a maintainer will have to merge. Many people are still coming back from time off, so it may just take some time to get merged in. I'll ping people next week if there's no update 👍

Sounds good, thanks for the PR and the follow-ups! 🎉

**Review by dmitryax — APPROVED** [review]

**Comment by dmitryax** [comment]
@crobert-1 , can you please apply the same fix in core? mdatagen is being moved there

**Comment by crobert-1** [comment]
> @crobert-1 , can you please apply the same fix in core? mdatagen is being moved there

Done: https://github.com/open-telemetry/opentelemetry-collector/pull/9229

**Pull request merged**
